### ISSUE TITLE
Default "modern" font on macOS doesn't include italics

### ISSUE THREAD

**Issue by achabotl** [enthought/chaco#521]
**Problem Description**

The default font "modern" on macOS is Lucida Grande, which does not include italics. It makes it feels like there's a bug in Chaco. (In fact there's a bug in Kiva (https://github.com/enthought/enable/issues/404), but that's a separate issue).

xref: https://github.com/enthought/enable/issues/406
**Reproduction Steps:**

```python
from numpy import linspace, pi, sin, cos
from traits.api import HasTraits, Instance
from traitsui.api import View, UItem
from chaco.api import Plot, ArrayPlotData
from enable.api import ComponentEditor
from kiva.api import ITALIC


class PlotWithModifiedLegend(HasTraits):
    plot = Instance(Plot)

    def _plot_default(self):
        x_array = linspace(0.0, 2*pi)
        data = ArrayPlotData(x=x_array, sin=sin(x_array), cos=cos(x_array))
        plot = Plot(data)
        plot.plot(('x', 'sin'), color='blue', name='sine')
        plot.legend.font.style = ITALIC
        plot.legend.visible = True
        return plot

    traits_view = View(
        UItem('plot', editor=ComponentEditor()),
    )

plot_with_legend = PlotWithModifiedLegend()
plot_with_legend.configure_traits()

```

**Expected behavior:**

The default "modern" font should have at least regular, italic, bold, and bold italic styles.

The legend should be in italic.

**OS, Python version:** 
- Python 3.6.9 -- Enthought, Inc. (x86_64)
- macOS 10.15.5

**Comment by corranwebster** [comment]
Getting Lucida Grande as the default "modern" font feels like a double failure: on wxPython "modern" is supposed to give a monospaced sans-serif font like Andale Mono, Consolas or Monaco (as opposed to "typewriter" which gives a monospaced serif font like Courier).  That's implying that whatever default for "modern" is is not commonly available on OS X.

Lucida Grande was the system font on OS X <= 10.9, which was probably why it chosen.  The default system font on OS X is now Helvetica Neue, which does have bold and italic options.

**Comment by jwiggins** [comment]
Fixed in Enable via enthought/enable#698
